src/viam/components/movement_sensor/movement_sensor.py

```diff
@@ -108,24 +111,34 @@ async def get_readings(self, *, extra: Optional[Dict[str, Any]] = None, timeout:
                 altitude: float,
                 linear_velocity: Vector3,
                 angular_velocity: Vector3,
+                linear_acceleration: Vector3
                 compass: float,
                 orientation: Orientation
             }
         """
-        ((pos, alt), lv, av, la, comp, orient) = await asyncio.gather(
+        (pos, lv, av, la, comp, orient) = await asyncio.gather(
             self.get_position(extra=extra, timeout=timeout),
             self.get_linear_velocity(extra=extra, timeout=timeout),
             self.get_angular_velocity(extra=extra, timeout=timeout),
             self.get_linear_acceleration(extra=extra, timeout=timeout),
             self.get_compass_heading(extra=extra, timeout=timeout),
             self.get_orientation(extra=extra, timeout=timeout),
+            return_exceptions=True,
         )
-        return {
-            "position": pos,
-            "altitude": alt,
-            "linear_velocity": lv,
-            "angular_velocity": av,
-            "linear_acceleration": la,
-            "compass": comp,
-            "orientation": orient,
-        }
+
+        readings = {}
+        if not isinstance(pos, GRPCError):
```

Comment:
I'd like these checks to be affirmative - so checking that the returned values is of the type expected, instead of it not being an exception.

Reply:
How it currently works in RDK:

* For a local sensor, it will check if the error is "Unimplemented." If yes, it will add a default value. If not, it will raise the error
* For a remote sensor (using the client), it will raise all errors, including "Unimplemented" ones

My understanding is that if we want to somewhat mimic RDK functionality (but with key omissions), then we should check if this error is either `NotImplementedError` or `viam.errors.MethodNotImplementedError`. If yes, we **OMIT** the value from the dict. If it is any other type of error, we should raise the error. 

@cheukt please confirm?

Reply:
I think your understanding is fair - just have to explicit that it could be brittle

Reply:
I think this is the same level of brittleness as in RDK

Reply:
[nit], but non-blocking - maybe we should stick the brittle logic into a function like we having in RDK where we do a similar string-based check? https://github.com/viamrobotics/rdk/blob/main/robot/client/client.go#L112-L114